### ISSUE TITLE
Collapsing module makes UI temporarily unresponsive

### ISSUE THREAD

**Issue by dterrahe** [darktable-org/darktable#12700]
While testing #12637 (and #12691) I noticed that collapsing an iop module (for example by expanding another one) often leads to the ui becoming unresponsive for a while seemingly waiting for the "working..." to finish.

It seems to be waiting for a mutex in `dt_image_update_final_size`.

Just highlighting this to those who have a deeper understanding of threading issues (@ralfbrown?) in case they are interested in digging into this or see something trivial. Maybe this update could be triggered by a pipe-finished signal?

Current master on debian and windows. Both opencl and no. Haven't done a bisect.

**Comment by gi-man** [comment]
I'm having a very similar issue. The Working... shows up and dt becomes unresponsive. If I then try to switch to lighttable, i then have to terminate dt. Zoom in or out shows the Working..., but no issues after. Turning Raw Overexposure On/Off shows the Working... and then the issue.

Fedora 36 KDE X11 4.1.0~git630.1880a8aa via OSB

**Comment by dterrahe** [comment]
This issue describes only temporary unresponsiveness, while waiting for "working" to finish (which is usually, depending on the modules in use, a few seconds). If your problem is that you have to kill dt because it never comes back, then that is not so similar at all and you may be better off, if you can describe steps to reproduce, with opening a separate issue.

**Comment by gi-man** [comment]
You are correct, it was a different issue. I started a new issue https://github.com/darktable-org/darktable/issues/12701
